client/tx/search.go

```diff
@@ -139,45 +139,45 @@ func FormatTxResults(cdc *codec.Codec, res []*ctypes.ResultTx) ([]Info, error) {
 // Search Tx REST Handler
 func SearchTxRequestHandlerFn(cliCtx context.CLIContext, cdc *codec.Codec) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		tag := r.FormValue("tag")
-		if tag == "" {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte("You need to provide at least a tag as a key=value pair to search for. Postfix the key with _bech32 to search bech32-encoded addresses or public keys"))
+		var tags []string
+		err := r.ParseForm()
+		if err != nil {
+			utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not parse query parameters", err.Error()))
 			return
 		}
-
-		keyValue := strings.Split(tag, "=")
-		key := keyValue[0]
-
-		value, err := url.QueryUnescape(keyValue[1])
-		if err != nil {
-			utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not decode address", err.Error()))
+		if len(r.Form) == 0 {
+			utils.PostProcessResponse(w, cdc, "[]", cliCtx.Indent)
 			return
 		}
 
-		if strings.HasSuffix(key, "_bech32") {
-			bech32address := strings.Trim(value, "'")
-			prefix := strings.Split(bech32address, "1")[0]
-			bz, err := sdk.GetFromBech32(bech32address, prefix)
+		for key, values := range r.Form {
+			value, err := url.QueryUnescape(values[0])
 			if err != nil {
-				utils.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+				utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not decode query value", err.Error()))
 				return
 			}
 
-			tag = strings.TrimRight(key, "_bech32") + "='" + sdk.AccAddress(bz).String() + "'"
+			if strings.HasSuffix(key, "_bech32") {
```

Comment:
What is this special handling we're doing for such a query key? Do you have an example of such a query?

Also, can we move this to a helper function with a proper godoc?

Reply:
sender_bech32=<accAddr>&receiver_bech32=<accAddr> to get all txs related to a specific address is one example

Reply:
Another option is to keep a static map of which tag keys map to address values (and so should be bech32-decoded in the URL parsing). I think that would be OK too.

Reply:
I like the map idea. Why not just `sender` and `receiver`?

Reply:
Created #2871